pythonforandroid/distribution.py

```diff
@@ -83,23 +93,62 @@ def get_distribution(cls, ctx, name=None, recipes=[],
 
         name_match_dist = None
 
+        req_archs = [arch.arch for arch in ctx.archs]
+
+        recipes_with_version = cls.get_recipes_with_version(ctx, recipes)
+
         # 0) Check if a dist with that name already exists
         if name is not None and name:
-            possible_dists = [d for d in possible_dists if d.name == name]
+            possible_dists = [
+                d for d in possible_dists if d.name.startswith(name)
+            ]
             if possible_dists:
                 name_match_dist = possible_dists[0]
 
         # 1) Check if any existing dists meet the requirements
         _possible_dists = []
         for dist in possible_dists:
-            if (
-                ndk_api is not None and dist.ndk_api != ndk_api
-            ) or dist.ndk_api is None:
+            if any(
+                [
+                    dist.ndk_api != ctx.ndk_api,
+                    dist.android_api != ctx.android_api,
+                    set(dist.archs) != set(req_archs),
+                    dist.ndk_api is None,
+                ]
+            ):
                 continue
-            for recipe in recipes:
-                if recipe not in dist.recipes:
+            checked_dist_recipes = set()
```

Comment:
same here for string literals `{}` which by the way is slightly faster in Python (because `set` would require alias resolution), but who cares :smile:

Reply:
Nevermind ref https://github.com/kivy/python-for-android/pull/1926#discussion_r308382952
Thanks @opacam